t/integration/test_events.py

```diff
@@ -0,0 +1,85 @@
+import asyncio
```

Comment:
Why all the use of asyncio in these tests instead of being more similar to the existing tests?